go-controller/pkg/util/ndp/na.go

```diff
@@ -0,0 +1,104 @@
+package ndp
+
+import (
+	"fmt"
+	"net"
+	"net/netip"
+
+	"github.com/mdlayher/ndp"
+	"golang.org/x/net/ipv6"
+
+	"k8s.io/klog/v2"
+)
+
+// NeighborAdvertisement represents a Neighbor Advertisement for an IPv6 address.
+type NeighborAdvertisement interface {
+	// IP returns the IPv6 address
+	IP() net.IP
+	// MAC returns the MAC address to advertise (nil means use interface MAC)
+	MAC() *net.HardwareAddr
+}
+
+type neighborAdvertisement struct {
+	ip  net.IP
+	mac *net.HardwareAddr
+}
+
+// NewNeighborAdvertisement creates a new Unsolicited Neighbor Advertisement with validation that the IP is IPv6.
+func NewNeighborAdvertisement(ip net.IP, mac *net.HardwareAddr) (NeighborAdvertisement, error) {
+	if ip.To4() != nil {
+		return nil, fmt.Errorf("only IPv6 addresses can be used for NeighborAdvertisement, got IPv4 %s", ip.String())
+	}
+	if ip.To16() == nil {
+		return nil, fmt.Errorf("only IPv6 addresses can be used for NeighborAdvertisement, got %s", ip.String())
+	}
+	if ip.IsMulticast() || ip.IsUnspecified() {
```

Comment:
is this check for multicast correct? i thought neighbor advertisement uses multicast

Reply:
It uses multicast as the destination, not the address we send the NA for.